Create a page for explaining what Tizen IoT platform is (#1194)
Samsung/tizen-docs: docs/platform/what-is-tizen/profiles/iot.md

# Tizen IoT

Tizen IoT profile allows you to make various kinds of IoT devices and develop applications for the IoT devices.
Tizen IoT Headed and Headless binaries support any type of IoT devices based on the Linux kernel.

![iot_building_block](media/iot_building_block.png)

> ## ABB: API building block

Reviewer: Suggestion: Could we remove "ABB:" and just say API building block

> Tizen IoT provides a building block pool of components mainly based on the [Tizen native API sets](../../../application/native/api/wearable/5.5/index.html).

Author: thanks for letting me know :)

> Tizen IoT provides a building block pool of components mainly based on the [Tizen native API sets](../../../application/native/api/wearable/5.5/index.html).

Author: done :)

Custom IoT images can be created by combining the building blocks in various ways as per the requirements.

Required packages are installed based on the dependencies related to the specified building blocks.

## Types of building block presets

### IoT core
- The minimum set of bootable modules that make up the Tizen platform image for IoT devices.
- The preset contains core components that are required for a basic system image to boot.

### Headless
- The set of modules that make up the Tizen platform image for headless IoT devices without a display.
- The preset contains IoT core presets with other components specifically required for headless image.

> - Example of specific components for headless image includes Audio moduldes, Bluetooth modules, and so on.

Author: done :)

### Headed
- The set of modules that make up the Tizen platform image for headed IoT devices with a display.
- The preset contains IoT core presets with other components specifically required for headed image.
- Example of specific components for headed image includes Notification API, graphic modules, and so on.

## MIC image creator

MIC is an image creator that is used to create images for Tizen. 

The tool offers three major functions as follows:
- create an image in a specific format
- chroot into an image
- convert an image to another format

> For more information on the tool and its functions, see [MIC Image Creator](https://docs.tizen.org/platform/reference/mic/mic-overview/).

Author: done :)

In this document, we will focus on creating the image using MIC.

## Create image using MIC

Before you create an image using MIC, ensure that the following are supported:
- OS: Ubuntu 14.04 LTS or higher

> - Git Build System(GBS) and Image Creator(MIC) for Tizen developers

Reviewer: Global: Please provide a space before the brackets. ALso, please provide a ":" at the end of the sentence:
Question is it Git Build System or git-build-system?  Reference: https://source.tizen.org/documentation/reference/git-build-system

```suggestion
- git-build-system (GBS) and Image Creator (MIC) for Tizen developers:
```

Author: done :)

    ```shell
    $ echo "deb [trusted=yes] http://download.tizen.org/tools/latest-release/Ubuntu_16.04/ /" | sudo tee /etc/apt/sources.list.d/tizen.list
    $ sudo apt-get update
    $ sudo apt-get install gbs mic
    ```

To create a custom Tizen IoT image, follow these steps:

> 1. Download a kickstart(.ks) file from the [Tizen download site](http://download.tizen.org/snapshots/tizen/unified/). 

Reviewer: ```suggestion
1. Download a kickstart (.ks) file from the [Tizen download site](http://download.tizen.org/snapshots/tizen/unified/).
```

Author: done :)

    - You may choose the file for the binary that is most similar to the custom image you are trying to create.

2. Edit the kickstart file in accordance with the requirements of the custom image. 

>     - You can add the name of the required building blocks under the %packages section of the kickstart file.

Reviewer: Please add a colon instead of a period as it is followed by a code.

- You can add the name of the required building blocks under the %packages section of the kickstart file:

Author: done :)

        ```shell
        %packages

        # @ IoT Headless Base
        building-blocks-root-Preset_iot_core
        # @ IoT Adaptation RPI
        building-blocks-sub1-Preset_img_headless-rpi3
        # Others
        # camera blocks  <-- New building block to use camera feature added
        building-blocks-sub2-Preset_boards-RPI3-Camera
        
        %end
        ```

    - To find the name of a specific building block, see the [mapping-bb-rs.xml](https://git.tizen.org/cgit/tools/building-blocks/plain/mapping-bb-rs.xml?h=tizen) file in the [Tizen building block repository](https://git.tizen.org/cgit/tools/building-blocks).

> 3. Create image with the git-build-systemm(gbs) command.
> 
>     ```shell
> 	$ gbs createimage --ks-file=YOUR_CUSTOM.ks
> 	```
>     - The new custom image can be found in the 'mic-output' directory.
>         ```shell
> 	    $ ls -al ./mic-output
> 	    ```

Reviewer: Suggest changing to:

```suggestion
3. Create image with the git-build-system (gbs) command:

    ```shell
	$ gbs createimage --ks-file=YOUR_CUSTOM.ks
	```
    The new custom image can be found in the 'mic-output' directory:
        ```shell
	    $ ls -al ./mic-output
	    ```
```

Please verify

Author: done :)

        
For more information, see [Creating Tizen Images with MIC](../../developing/creating.md).

> 

Reviewer: Please remove extra lines.

Author: done :)

> ## Developing applications for Tizen IoT

Author: done :)

> # Tizen IoT

Author: i have changed the structure of the documents in regard to your concern :)

Tizen IoT profile allows you to make various kinds of IoT devices and develop applications for the IoT devices.
Tizen IoT Headed and Headless binaries support any type of IoT devices based on the Linux kernel.

![iot_building_block](media/iot_building_block.png)

## ABB: API building block

Tizen IoT provides a building block pool of components mainly based on the Tizen native API sets.

|Tizen Native API Sets|
|-----------------|
|Account Native API Group|
|Application Framework Native API Group|
|Base API Group|
|Content API Group|
|Context API Group|
|Location API Group|
|Machine Learning API Group|
|Maps API Group|
|Messaging API Group|
|Multimedia API Group|
|Network API Group|
|Security API Group|
|Social API Group|
|System API Group|
|Telephony API Group|
|UI API Group|
|UIX API Group|
|Web API Group|

> Custom IoT images can be created by combining the building blocks in various ways according to the needs.

Author: done :)

> Required packages will be installed based on the dependency related to the specified building blocks.

Author: done :)

> ## Types of Building Block Presets

Author: done :)

### IoT Core
- The minimum set of bootable modules that make up the Tizen platform image for IoT devices.
- The preset contains core components that are required for a basic system image to boot.

### Headless
- The set of modules that make up the Tizen platform image for headless IoT devices without a display.
- The preset contains IoT core presets with other components specifically required for headless image.

> - Example of specific components for headless includes Audio moduldes, Bluetooth modules, and etc.

Author: done :)

### Headed
- The set of modules that make up the Tizen platform image for headed IoT devices with a display.
- The preset contains IoT core presets with other components specifically required for headed image.

> - Example of specific components for headed includes Notification API, graphic modules, and etc.

Author: done :)

> 

Reviewer: Please remove extra line.

Author: done :)

> ## MIC Imager Creator

Reviewer: Please change Imager to Image:

## MIC Image Creator

Author: done :) + sentence case

MIC is an image creator that is used to create images for Tizen. 

The tool offers three major functions as follows:
- creating an image in a specific format
- chrooting into an image

> - coverting an image to another format

Reviewer: Suggest using verb form:

- create an image in a specific format
- chroot into an image
- covert an image to another format

Reviewer 2: One doubt to the last list item.

Would you please check if it is "covert" or "convert"?

Reviewer: Thank you for pointing this out. it should be convert.

Suggest using verb form:

- create an image in a specific format
- chroot into an image
- convert an image to another format

Author: done :)

> For more information about the tool and its functions, refer to [MIC Image Creator](https://docs.tizen.org/platform/reference/mic/mic-overview/).

Author: done :)

> In this document, we will focus on creating the image using MIC.

Reviewer 2: I think an empty line is needed between the last list item (Git Build System...) and the code block, in your suggestion.
For now the code block layout is broken and is displayed as a code style.

> In this document, we will focus on creating the image using MIC.

Author: done :)

To create a custom Tizen IoT image, 

0) Check prerequisites
    - OS: Ubuntu 14.04 LTS or higher
    - Git Build System(GBS) and Image Creator(MIC) for Tizen developers
        ```shell
        $ echo "deb [trusted=yes] http://download.tizen.org/tools/latest-release/Ubuntu_16.04/ /" | sudo tee /etc/apt/sources.list.d/tizen.list
        $ sudo apt-get update
        $ sudo apt-get install gbs mic
        ```

> 1) Download a kistart(.ks) file from the [Tizen download site](http://download.tizen.org/snapshots/tizen/unified/). 

Reviewer: Global: Is it Kickstart file? Please clarify

Download a Kickstart (.ks) file from the [Tizen download site]

Author: done ;)

>     - You may choose the file for the binary that is most similar to the custom image you are trying to create.

Author: done :)

2) Edit the kickstart file in accordance with the requirements of the custom image. 

>     - You can add the name of the required building blocks under the %pakckages section of the kickstart file.

Author: done :)

        ```shell
        %packages

        # @ IoT Headless Base
        building-blocks-root-Preset_iot_core
        # @ IoT Adaptation RPI
        building-blocks-sub1-Preset_img_headless-rpi3
        # Others
        # camera blocks  <-- New building block to use camera feature added
        building-blocks-sub2-Preset_boards-RPI3-Camera
        
        %end
        ```

>     - To find the name of a specific building block, refer to the [mapping-bb-rs.xml](https://git.tizen.org/cgit/tools/building-blocks/plain/mapping-bb-rs.xml?h=tizen) file in the [Tizen building block repository](https://git.tizen.org/cgit/tools/building-blocks).

Author: done :)

> 3) Create image with the gbs command.

Reviewer: Suggestion:

3. Create image with the git-build-systemm (gbs) command:

Author: done :)

    ```shell
	$ gbs createimage --ks-file=YOUR_CUSTOM.ks
	```
    - The new custom image can be found in the 'mic-output' directory.
        ```shell
	    $ ls -al ./mic-output
	    ```

> For more information, refer to [Creating Tizen Images with MIC](https://source.tizen.org/documentation/developer-guide/getting-started-guide/creating-tizen-images-mic).

Reviewer 2: I accepted @surbhig12's comment, so updated my previous comment.

> For more information, refer to [Creating Tizen Images with MIC](https://source.tizen.org/documentation/developer-guide/getting-started-guide/creating-tizen-images-mic).

Author: done :)

> ## Developing Applications for Tizen IoT

Author: done :)

> To develop applications for the IoT profile:

Reviewer: Suggest providing some more information then a leading line, as the heading itself is serving as a leading line.

Author: How about 
For more information on developing applications for the IoT profile, see [Create IoT .NET Application](../../../application/dotnet/get-started/iot/first-app.md).
?

Reviewer: This is fine but providing some extra information may help.

Author: the same sentences with the same level of information are used on the pages for all other platforms.

> - [IoT .NET application](../dotnet/get-started/iot/first-app.md)

Author: done :)

# Tizen IoT

Tizen IoT profile allows developers to make various kinds of IoT devices and develop applications for the devices.
Tizen IoT Headed and Headless binaries support any type of IoT devices based on the Linux kernel.

> ## Development Environment for Tizen IoT

Reviewer: Global: Please use sentence case for all the headings.

Line 6: ## Development environment for Tizen IoT
Line 18: ## Create Tizen IoT images

Author: I also made it consistent with the other pages. Could you confirm once more?

Tizen IoT supports the following environment for development:
- Hardware targets
    - Raspberry Pi 3 and 4 as the official reference boards
- Platform images
    - Headless image for IoT devices without a display
    - Headed image for IoT devices with a display

## Developing Applications for Tizen IoT
To develop applications for the IoT profile:

> - [Create Your First Tizen IoT .NET Application](../../../application/dotnet/get-started/iot/first-app.md)

Reviewer: Just a suggestion:
Could we add this point as a single sentence and not as a single bullet:

To develop applications for the IoT profile, see [Create Your First Tizen IoT .NET Application](../../../application/dotnet/get-started/iot/first-app.md).

Please ignore if invalid.

Author: I think it would be better to have a consistent style with other pages (mobile and wearable) 
+ we could add more bullet points when more information gets available.

- The minimum set of bootable modules that make up the Tizen platform image for IoT devices.
- The preset contains core components that are required for a basic system image to boot.

**Building block prsets for Headless:**
- The set of modules that make up the Tizen platform image for headless IoT devices without a display.
- The preset contains IoT core presets with other components specifically required for headless image.
- Example of specific components for headless image includes Audio modules, Bluetooth modules, and so on.

**Building block prsets for Headed:**
- The set of modules that make up the Tizen platform image for headed IoT devices with a display.
- The preset contains IoT core presets with other components specifically required for headed image.
- Example of specific components for headed image includes Notification API, graphic modules, and so on.

Custom IoT images can be created by combining API building blocks in various ways as per the requirements. Required packages are installed based on the dependencies related to the specified building blocks.

![iot_building_block](media/iot_building_block.png)

To create images for the IoT profile:

> - [Creating Tizen Images with MIC](../../developing/creating.md)

Reviewer: Just a suggestion:
Could we add this point as a single sentence and not as a single bullet:

To create images for the IoT profile, see [Creating Tizen Images with MIC](../../developing/creating.md).

Please ignore if invalid.

Author: I think it would be better to have a consistent style with other pages (mobile and wearable).
Also, we could add more bullet points when more information gets available.
